### ISSUE TITLE
Make API error message more verbose

### ISSUE THREAD

**Issue by sahildua2305** [sahildua2305/hackIDE#32]
I think we should make the error message more verbose explaining that it's API's fault and it's temporary.

For more details: See #31

**Comment by bronwynbiro** [comment]
I'm new to open source but would like to work on this if that's okay! Do we want to simply say the error is from Hackerearth API and is temporary or do we want to show the status detail that the error returns, i.e. "correct the compilation/syntax errors"?

**Comment by sahildua2305** [comment]
@bronwynbiro sorry for late reply. Are you still interested in tackling this issue?

**Comment by lawfets** [comment]
It would be my first time, but with some help I think I should manage

**Comment by sahildua2305** [comment]
@lawfets thanks for your interest in contributing to HackIDE.

Regarding this issue, it should be an easy one. We want to show better error messages to users in case API calls to HackerEarth fails. Please look into the main javascript file to find `fail` methods in ajax requests.

**Comment by lawfets** [comment]
I'll check it out

**Comment by lawfets** [comment]
where can I find the main javascript file?

**Comment by sahildua2305** [comment]
Here - https://github.com/sahildua2305/hackIDE/blob/master/hackIDE/static/hackIDE/js/custom.js#L309


This is the error message we show if API request fails. We should be able to see the response and show the message according to the response message.

**Comment by lawfets** [comment]
what should I write?
This is an API request failure, this should only be temporary.
{"errors": {}, "code_id": "6879d1z", "message": "OK", "run_status": {"status": "CE", "status_detail": "Correct the compilation/syntax errors."}, "compile_status": "Error code: 1200", "web_link": 

btw how do I adapt the file, do i fork it and than make a pull request?

**Comment by sahildua2305** [comment]
>what should I write?

Depends. Can we see what error message is returned from the API request. It may also be a timeout. You will need to investigate how we can check what caused an Ajax request to fail.

>btw how do I adapt the file, do i fork it and than make a pull request?

Yes, right!

**Comment by lawfets** [comment]
I don't think I can do anything at the moment as the original bug seems to be fixed and hacker earth has no bug with code_id": "6879d1z. So I cannot replicate any failures.

**Comment by lawfets** [comment]
I did find someone else with equal problem on github with a compiler but no answer there.
https://github.com/codehimanshu/compiler/issues/4  also in connection with Hackerearth

**Comment by lawfets** [comment]
Is there anything I'm missing I should do to try to solve this issue

**Comment by sahildua2305** [comment]
@lawfets you need to do some search regarding what kind of response do we get when Ajax request fails. You will need to read a bit about Ajax for this. Accordingly, we can then handle the error cases.

**Comment by adi8** [comment]
I would be interested in working on this issue. Is this issue assigned to @lawfets ?

**Comment by lawfets** [comment]
feel free to hop in adi8, I'm still in learning mode for AJAX so at the moment, nothing I can do

**Comment by adi8** [comment]
Cool. I have a pretty good idea of what to do. Would like to know from @sahildua2305 what is exactly expected. You see I'm also a new to open source projects.

**Comment by sahildua2305** [comment]
@adi8 Check this issue - https://github.com/sahildua2305/hackIDE/issues/31#issuecomment-234212904

Should be an easy fix.

**Comment by adi8** [comment]
This issue is very difficult to replicate. Based on the info that I could gather on HackerEarth API documentation(https://www.hackerearth.com/docs/api/developers/code/legacy/) and the #31 issue I made some changes in javascript. I'll issue a pull request.

**Comment by Bilalharoon** [comment]
I am new to open source but I would like to contribute

**Comment by Nayan-Das** [comment]
@sahildua2305 I want work on this issue it is not already solve.Should I take this up?

**Comment by manavendrasen** [comment]
I would like to work on this is this still open?

**Comment by MohammedBajuaifer** [comment]
Would like to work on this issue, is it still open?

**Comment by sahildua2305** [comment]
Yes, after staying on snooze for a while, it's up for grabs again.

**Comment by chandhu455** [comment]
@sahil865gupta  can i look into this issue? Is this still valid one?

**Comment by parimalakettymuthu** [comment]
@sahildua2305  -- Is this issue still open ?

**Comment by soham-sagade** [comment]
is this open?

**Comment by Sophyia7** [comment]
Hello, is this issue still open? I am new to contributing and would love to take it on?

**Comment by 0xClint** [comment]
Is this issue solved?
